src/kind.ts

```diff
@@ -0,0 +1,79 @@
+import { Job } from "@brigadecore/brigadier";
+
+export const kindJobImage = "radumatei/golang-kind:1.11-0.4";
+
+export class KindJob extends Job {
+    constructor(name: string, image?: string) {
+        if (image == undefined) {
+            image = kindJobImage;
+        }
+        super(name, image);
+
+        // kind needs to run as a privileged pod
+        this.privileged = true;
+
+        // since the cluste creation takes some time,
+        // set the timeout to 30 minutes
+        this.timeout = 180000;
+
+        // kind needs to add the following volumeMounts to function properly
+        // the Brigade project must enable `allowHostMounts`
+        this.volumes = [
+            {
+                name: "modules",
+                hostPath: {
+                    path: "/lib/modules",
+                    type: "Directory"
+                }
+            },
+            {
+                name: "cgroup",
+                hostPath: {
+                    path: "/sys/fs/cgroup",
+                    type: "Directory"
+                }
+            },
+            {
+                name: "docker-graph-storage",
+                emptyDir: {}
+            }
+        ];
+        this.volumeMounts = [
+            {
+                name: "modules",
+                mountPath: "/lib/modules",
+                readOnly: true
+            },
+            {
+                name: "cgroup",
+                mountPath: "/sys/fs/cgroup"
+            },
+            {
+                name: "docker-graph-storage",
+                mountPath: "/var/lib/docker"
+            }
+        ];
+
+        // to add your own tasks to this job, use job.tasks.push()
+        this.tasks = [
+            // when the pod finishes, regardless of the exit code, delete the cluster
+            // to avoid resource leaks
+            // see https://github.com/kubernetes-sigs/kind/issues/759
+            "trap 'kind delete cluster' SIGTERM EXIT",
```

Comment:
I'm still unsure on whether or not we need to capture the exit code associated with the signal that comes in, in order to properly exit the container.  Otherwise, the job task that failed might be lost when `kind delete cluster` exits zero.  Naturally, we also want the job to fail if `kind delete cluster` fails, so perhaps it isn't as simple as always exiting with the captured exit code that comes in... worthy of some testing, for sure.

Reply:
Consider the following:

```javascript
const { events, Job } = require("brigadier");

events.on("exec", () => {
  var job = new Job("it-a-trap-fail", "alpine:3.8");
  job.tasks = [
    "trap 'echo trapping sigterm and exit' SIGTERM EXIT",
    "sleep 10",
    "exit 1"
  ];

  job.run();
});
```

Letting this finish results in the build failing as expected because of `exit 1`.

Reply:
That being said, you might be right, trapping `EXIT` handles both normal ending of the script (regardless of exit code), and Kubernetes terminating the pod.

Reply:
Yes, I believe we just need `EXIT`.